### DISCUSSION TITLE
Welcome to ME3255 Summer session!

### DISCUSSION THREAD

**Discussion by cooperrc** [cooperrc/computational-mechanics#241] [General]
Greetings!

Thanks for joining our Computational Mechanics Summer session. I'm excited to start building Python solutions to engineering problems and exercise our computational thinking. This is one of my favorite courses in our Mechanical Engineering department. We get to really test our engineering theories and predictions with models, visualizations, and numerical results. 

To get started, I want you to have a GitHub account and join this discussion. As we continue, post bugs, error codes, or even "I don't know where to start" questions and we'll all chip in answers. 

The first questions for our learning community:

1. What is one reason you're in this class (outside of requirements)?
2. What is your favorite Summer activity?
3. Share a snippet of Python code and tell us what it does\*

> \* use ` ```python ``` ` brackets to make a copy-able code cell e.g. 
>
>````
>```python
>print('hello world')
>```
>````
>
>gives us
>
>```python
>print('hello world')
>```

*Note: This is a public forum so only share what you're comfortable having online. Use email or direct messages for any private/personal discussion topics. Also, you can edit/delete any messages you create here if you need.*

**Comment by cooperrc** [comment]
> 1. What is one reason you're in this class (outside of requirements)?

I struggled with Matlab when I was an undergraduate at Drexel University. When I started graduate school, my advisor took the time to show me how I could start solving difficult inverse problems with finite element analysis (FEA) and Matlab. One example of an inverse problem would be:

> If I know the length and cross-section of this beam and boundary conditions, I can calculate its deflection, but what if I want to change its cross-section until it has a given deflection of 2 mm?
> 
> I start with a guess, give it an area of 10 mm^2 and find it deflects 4 mm. Now, I plug in a new area, maybe 11 mm^2 and find it deflects 3.2 mm. 

This process felt so freeing, I didn't need complicated algebra and calculus to invert every equation I had, I could use a computer to guess-and-check my answers for me. 

> 2. What is your favorite Summer activity?

I love the beach in the Summer. We (my 6-yo, 8-yo, wife and me) usually have x3 beaches we have to hit, 
- Niantic, CT 
- Robert Moses, NY / Jones Beach, NY
- Watch Hill, RI

> 3. Share a snippet of Python code and tell us what it does*

```python
import numpy as np
import matplotlib.pyplot as plt
plt.style.use('fivethirtyeight')

g = 9.81 # m/s/s
L = 0.5 # m
w = np.sqrt(g/L)
t = np.linspace(0, 2*np.pi/w)

theta = np.pi/6*np.cos(w*t)


plt.plot(t, theta*180/np.pi)
```

This is a quick piece of code to plot the angle of pendulum that is released 30-degrees from its resting position at 0-degrees. I import x2 libraries: numpy and matplotlib to make arrays and plot them and I also set the plot style to one I like, ['fivethirtyeight'](https://matplotlib.org/stable/gallery/style_sheets/style_sheets_reference.html)

I set up its length `L` gravity of Earth `g` and the time going for one oscillation, $t = [0...\frac{2\pi}{\omega}]$ where $\omega = \sqrt{g/L}$ is the natural frequency of the pendulum. Then, I calculate the linear solution where $\theta(t) = 30^o\cos\omega t = \frac{\pi}{6}\cos \omega t$ and plot with `plt.plot`

**Comment by adimakarla** [comment]
**What is one reason you're in this class (outside of requirements)?**

Hello, my name is Aditya Makarla and I am a rising senior studying Mechanical Engineering and German in UCONN's Eurotech program. One reason that I am taking this class is to learn more about mechanical systems in addition to getting a little more experience in Python. 

**What is your favorite Summer activity?**

During the Summer, I really enjoy travelling around, whether it is to another state or another country. Being from a small state like CT, it is even more exciting when I get to go to new places that are bigger and have a lot to offer in terms of activities, sights, etc. I would like to explore more of Europe in the future. 

**Share a snippet of Python code and tell us what it does:**

Here is a code which simply calculates the vertical displacement of an object in free fall with set initial conditions for the velocity and time:

```python
def vertical_displacement(v, t, a=9.81):
    y = v * t + 0.5 * a * t**2
    return y

# Sample initial conditions
initial_velocity = 0  # initial vertical velocity in m/s
time = 6  # seconds

displacement = vertical_displacement(initial_velocity, time)
print(f"The vertical displacement after {time} seconds is {displacement} meters.")
```

**Reply by cooperrc** [reply]
What's your favorite destination so far? In Europe, I loved Croatia. My family and I flew into Split, then sailed for a week before I started my PhD program. 

The nice part about your code is you can give it an `array` of time values and get an `array` of heights, `y`.

**Reply by adimakarla** [reply]
I have been to London, Paris, and Amsterdam so far; out of these 3, I enjoyed London the most because it reminds a lot of Boston. I really hope to visit Switzerland at some point within the next year as well as Italy down the line.

**Comment by Adam-Skarre** [comment]
**What is one reason you're in this class (outside of requirements)?**
Hello my name is Adam, I'm in this class to deepen my understanding of computational methods and enhance my ability to solve complex engineering problems using Python. This knowledge will be invaluable in my future career as a mechanical engineer... or as a programmer!
**What is your favorite Summer activity?**
My favorite summer hobby is kayaking or going to the beach with friends and family. Being in nature is a great way to recharge. 
**Share a snippet of Python code and tell us what it does***
```
import numpy as np

# Coefficient matrix
A = np.array([[3, 2], [1, 2]])

# Right-hand side vector
B = np.array([5, 5])

# Solving the system of equations
X = np.linalg.solve(A, B)

print("Solution:", X)
```
This code defines a system of linear equations in matrix form from AxX=B and solves for the vector X. The np.linalg.solve function is used to find the solution, which in this case will output the values of 
X that satisfy the equations represented by matrix A and vector B.

**Reply by cooperrc** [reply]
I love kayaking in the sound or even down in Mansfield Hollow. Do you prefer whitewater, lake, or ocean?

Love to see some Linear Algebra already. 

> maybe programmer

Ooh, are you interested in scientific computing, data science, or something else?

**Reply by Adam-Skarre** [reply]
I normally Kayak in lakes around me, and once I get to the bay I turn around! 

With programming, I am interested in software development. However, machine learning interests me even though it's different. As a whole, I'd like to sharpen my programming skills to implement it into machine learning, or software development.

**Reply by cooperrc** [reply]
Cool, we're going to do a little machine learning work with [Module_02](https://cooperrc.github.io/computational-mechanics/module_02/03_Linear_Regression_with_Real_Data.html) and [Module_04](https://cooperrc.github.io/computational-mechanics/module_04/03_Linear-regression-algebra.html). Talking to some data scientists in industry, they seem to prefer linear regression ML because its easier to interpret, debug, and deploy.

**Comment by ColeScottUconn** [comment]
**1. What is one reason you're in this class (outside of requirements)?**
Hello my name is Cole, I am taking Computational Mechanics this summer to learn more about how Python can be applied to solve engineering problems. In previous courses such as Linear Systems and Applied Measurements I found that using Python made calculations significantly more efficient and simpler compared to manual calculations. I am excited to learn how to utilize Python further to make engineering work more efficient.

**2. What is your favorite Summer activity?**
My favorite summer activities are hanging out at a lake and going to baseball games

**3. Share a snippet of Python code and tell us what it does**
```python
import math

# Given values
damping_coefficient = 50  # Ns/m
mass = 10  # kg
spring_constant = 200  # N/m

critical_damping = 2 * math.sqrt(mass * spring_constant)
damping_ratio = damping_coefficient / critical_damping

print(f"Damping ratio of the system: {damping_ratio:.2f}")
```
This python snippet calculates the damping ratio of a linear system using the equation: ζ = c / (2sqrt(km))

**Reply by cooperrc** [reply]
Awesome script, one tiny recommendation

~`import math`~ `import numpy as np`

[NumPy](numpy.org) is _way_ more powerful than the default `math` library. We'll use it a lot in this course and I'd recommend using it in almost every situation. 

I'm headed to a little league game in about 60min. It really does feel like summer now :)

**Comment by nebabebe** [comment]
Hi everyone! My name is Nebeyu, or you can also call me Neb. I am a student at George Mason University in VA. Currently going into my fourth year. 

**What is one reason you're in this class (outside of requirements)?**

I am in this class because I actually wanted to be exploited to more coding! I heard that this class has some coding that we can do, and I truly believe this will help benefit me in the future and help grow my skills with Python.

**What is your favorite Summer activity?**

My favorite summer activity is to hang out with my friends and do lots of car stuff together. We modify our cars and like to have fun with them, so it's a common thing for us to get together and install a new part or take some rolling shots of everyone's cars on the road. 

**Share a snippet of Python code and tell us what it does**
``` wait_list=[]

def add_name(name):
    wait_list.append(name)
    print(f'{name} has been added.')

while True:
    name = input("Enter a name to add to the wait list (or type 'quit' to stop):")
    if name.lower() =="quit":
        break
    add_name(name)

print("Current Waitlist:")

for name in wait_list:
    print(name) 
```
Pretty basic, however helpful for restaurants attempting to keep track of people entering a waitlist. This would be just a snippet which just allows people to enter themselves onto a waitlist, but more things to add would be a removing function when the person has been attended for in the restaurant, as well as an approximate waiting time.

**Reply by cooperrc** [reply]
Welcome to our UConn class! We'll definitely be building our Python skills and hopefully you'll hang on to your work as a reference/portfolio for the future. 

I like this script for restaurant use. It would be cool to add some kind of timer to keep track of how long people have waited and try to sort by wait time on each query. Maybe also party size?

**Comment by Zeyad-hs** [comment]
-What is one reason you're in this class (outside of requirements)?

Hello everyone, my name is Zeyad. I am a junior Mechanical Engineering student at George Mason University, VA. Besides the fact that this course is the last requirement for my senior project, one reason I am in this class is to learn more about finite element analysis through coding. I currently work at the Center of Collision Safety and Analysis (CCSA), where we conduct finite element analysis on vehicles to see how they react in certain crash tests.

-What is your favorite Summer activity?

Two of my favorite summer activities are bass fishing and taking back road rides on my motorcycle.

-Share a snippet of Python code and tell us what it does:

I haven’t really done much with Python, but since I just mentioned summer activities, here is a simple code that suggests an activity based on the weather:

```python

def suggest_activity(weather):
    if weather == "sunny":
        return "Go for a motorcycle ride!"
    elif weather == "rainy":
        return "Stay in and do your HW!"
    elif weather == "cloudy":
        return "Perfect time for Bass fishing."
    else:
        return "Check the weather forecast again!"

print(suggest_activity(current_weather))
```

**Reply by cooperrc** [reply]
Welcome to our UConn course! 

For the weather, if you really want to up your activity function, wrap in a [weather checker API](https://www.geeksforgeeks.org/python-find-current-weather-of-any-city-using-openweathermap-api/) e.g. 

```python
 Enter city name : Delhi
 Temperature (in kelvin unit) = 312.15
 atmospheric pressure (in hPa unit) = 996
 humidity (in percentage) = 40
 description = haze
```

**Comment by LucyLyttle** [comment]
Hi. I'm Lucy, a rising senior here at UConn.

**1. What is one reason you're in this class (outside of requirements)?**
I'm taking this course because, aside from it being a requirement, I know how important Python and coding skills in general have become engineering. I want to make sure I'm able to use programming to solve the kind of problems I might be faced with.

**2. What is your favorite Summer activity?**
I mostly like to stay inside and relax during the summer, reading and playing video games. But I also like going to the beach. For years my family would go to the beach in Maine every summer, at my grandparent's house on Drake's Island.

**3. Share a snippet of Python code and tell us what it does**
The below code will plot population growth over a span of 50 years, given an initial population of 1000 individuals and a growth rate of 5%.
```
import numpy as np
import matplotlib.pyplot as plt

p0 = 1000 # initial population
r = 0.05 # growth rate

t = np.linspace(0,50,50)
population = p0 * np.exp(r*t)

plt.plot(t, population, color='r')
plt.xlabel('Time (yrs)')
plt.ylabel('Population')
plt.title('Population Growth')

**Reply by cooperrc** [reply]
You must be reading ahead, we're doing a growth rate problem in [HW_01](https://cooperrc.github.io/computational-mechanics/module_01/HW_01.html#problems-part-3)

Maine is beautiful in the Summer. Hope you make back at some point.

**Comment by Kevin-Harmeling** [comment]
What is one reason you're in this class (outside of requirements)?
My name is Kevin and I am a rising Senior studying Mechanical Engineering, I was recently in your class last semester for Applied Mechanics II. I am in this class to deepen my knowledge about Python to understand how I can use for engineering purposes.
What is your favorite Summer activity?
My favorite summer activity is going to the beach, no matter where it is. I prefer being in Florida which is where I am going in a couple weeks, but Rhode Island beaches can be nice too sometimes.
Share a snippet of Python code and tell us what it does*
```python

def kinetic_energy(mass, velocity):

 
    return 0.5 * mass * velocity ** 2

# Example
mass = 10  # in kilograms
velocity = 5  # in meters per second
ke = kinetic_energy(mass, velocity)
print(f"The kinetic energy of the object is {ke} joules")`
``` 
This code is able to calculate kinetic energy of anything given a mass and a velocity.

**Reply by cooperrc** [reply]
Florida in the Summer? You're braver than me. I don't like to go south of Maryland once July/Aug hit. We used to live outside Knoxville, TN when I was a postdoc. Trying to run in 90-deg 95% humidity was deadly. 


Nice KE function. would be cool to take the magnitude of the velocity vector, 

e.g. 

```python

  v = np.sqrt(np.sum(velocity**2))

  return 0.5*m*v**2
```

**Comment by MuhammadSaqab** [comment]
**1.) What is one reason you're in this class (outside of requirements)?**
One reason I'm taking this class (ME 3255) is because I wanna improve my python skills. 
**2.) What is your favorite Summer activity?**
My favorite summer activity is going somewhere out of the country for vacation and exploring the different landscapes and culture, preferably a hot tourist location such as Cancun or Dubai. 
**3.) Share a snippet of Python code and tell us what it does.**
```python
import math

m = 20 
c = 4.186
delta_t = 10

Q = m*c*delta_t
print('The heat absorbed by the water is ' Q)
``` 
The above python function is able to calculate the heat absorbed by water given a mass value, specific heat, and a change in temperature.

**Reply by cooperrc** [reply]
Its definitely fun to check out a new culture/city. 

nice specific heat calculator!

**Comment by johannadatta** [comment]
What is one reason you're in this class (outside of requirements)?
Hi everyone! I'm Johanna and am taking this class to better my coding skills and become more familiar with FEA and how to use it in future engineering roles. 
What is your favorite Summer activity?
I love going on road trips with friends or family!
Share a snippet of Python code and tell us what it does*
```python
def preference_survey():
    # User is asked which they prefer [beach/lake]
    preference = input("Do you prefer going to the beach or going to a lake? Please type 'beach' or 'lake': ")
    
    # Print different response for each option
    if preference == 'beach':
        print("Nice, I went to the beach over the long weekend!")
    elif preference == 'lake':
        print("Have you ever played mermaids?")
    else:
        print("Hmm, please try again and respond with either 'beach' or 'lake'.")
    
# Run the survey
preference_survey()
```
My snippet asks users if they prefer the beach or a lake and responds based on their choice!

**Reply by cooperrc** [reply]
`Do you prefer going to the beach or going to a lake? Please type 'beach' or 'lake':`
beach

`Nice, I went to the beach over the long weekend!`

**Comment by jiovanni5** [comment]
**What is one reason you're in this class (outside of requirements)?**
I'm Jiovanni, a rising Senior studying Mechanical Engineering. I am in this class to apply my knowledge about lab testing to Python for further engineering study.

**What is your favorite Summer activity?**
My favorite summer activity is playing basketball outside with my friends.

**Share a snippet of Python code and tell us what it does**
This code calculates Potential Energy

```python
import numpy as np
import math
F = 10
x = 10
k = F/x
U = .5*k*(x**2)
print('Maximum potential energy =',np.max(U))
```

**Reply by cooperrc** [reply]
interesting, you're using force and displacement as inputs instead of the spring constant.

**Comment by tzamani310** [comment]
What is one reason you're in this class (outside of requirements)?
Hello, my name is Theo (she/her) I'm a junior mechanical engineer from George Mason University in Virginia. Aside from this being my last prerequisite for my senior design, it will be fun to sharpen up my coding skills, I'm not the strongest with coding but I can't wait to learn more Python skills this summer!

What is your favorite Summer activity?
I love having more time to work on music and going to the gym. It's hard to make time for either during the semester, so it feels nice to have less on my plate and to reconnect with my hobbies and what keeps me thriving!

Share a snippet of Python code and tell us what it does
. In the code below, you enter a number and it will return if the number entered is an even or odd number.
```python 
num= int(input("please enter a number: ")) # this will show the question
if (num % 2) == 0:
   print("{0} is even".format(num)) # this will return a number if it is even
else:
   print("{0} is odd".format(num)) # this will return a number if it is odd

**Reply by cooperrc** [reply]
The Summer is a great time to reconnect with hobbies, or build new habits.

**Comment by PhantomAlex1** [comment]
What is one reason you're in this class (outside of requirements)?
Hello, my name David I'm a senior at George Mason University studying mechanical engineering. I am taking this class to gain experience on how to put the theories and equations I've learned into practice outside of college.

What is your favorite Summer activity?
I would say my favorite summer activity is going out to hangout with friends at a lake house or beach.

Share a snippet of Python code and tell us what it does
This code calculates the weighted grade of a student. They just need to input their ID, Midterm Grade, and Final Exam Grade

```
ID = input('Enter the NetID of the student:')
Mid = float(input('Enter the score of midterm exam:'))
Final = float(input('Enter the score of final exam:'))
Final_Score = round(((Mid * (40/100)) + (Final * (60)/100)),1)

print(f'The final score of {ID} is {Final_Score}' )
```

**Reply by cooperrc** [reply]
Welcome to our UConn course!

**Comment by shiddhipatel** [comment]
**What is one reason you're in this class (outside of requirements)?**
Hello! My name is Shiddhi and I am a rising senior studying mechanical engineering with a minor in engineering management. Other than this class being a requirement, I wanted to further enhance my skills using python because I know I will have to code in future projects, especially for Senior design. So I wanted to be prepared and gain as much experience as I can using python.

**What is your favorite Summer activity?**
My favorite summer activity is to have picnics with my friends at the different beaches that are around my town.

**Share a snippet of Python code and tell us what it does**
The code below calculates the factorial of any number entered. Factorial is the product of all positive integers less than or equal to the number entered. 

```python
def factorial(n):
    if n == 0 or n == 1:
       return 1
    else: 
       return n * factorial(n-1)

number = int(input("Enter a number to calculate its factorial: "))
result = factorial(number)
print(f"The factorial of {number} is {result}")
```

**Reply by cooperrc** [reply]
Ooh, nice recursive programming!

**Comment by riddhipatel03** [comment]
**What is one reason you're in this class (outside of requirements)?**
Hi! My name is Riddhi and I am a upcoming senior studying mechanical engineering. The reason as to why I am in this class is I wanted to gain more coding experience and knowledge so I can use it in the future.

**What is your favorite Summer activity?**
One of my favorite summer activities is traveling with my friends to different locations.

**Share a snippet of Python code and tell us what it does***
The code calculates area of a circle given its radius.

```python
import math

def circle_area(radius): 
    return math.pi*radius**2

radius= float(input("Enter the radius of the circle:"))
area= circle_area(radius)
print(f"The area of the cirlce is {area:.2f}")
```

**Reply by cooperrc** [reply]
any planned locations?

**Comment by haarisnadeem** [comment]
**What is one reason you're in this class (outside of requirements)?**
Hello everyone my name is Haaris Nadeem and I am a rising senior mechanical engineering student at UConn. Other than requirements, I am taking this class to further my knowledge with coding in regards to mechanical engineering. I am decent at coding differential equations/statistics with MATLAB so I'm hoping by the end of this class I can have a better understanding of python. 

**What is your favorite Summer activity?**
Going to the beach and just rotting in the sun. 

**Share a snippet of Python code and tell us what it does**
This python code finds the maximum value in a list.
```python
def find_max(numbers):
    max_value = numbers[0]
    for number in numbers:
        if number > max_value:
            max_value = number
    return max_value

print(find_max([3, 5, 2, 4, 6]))  

```

**Reply by cooperrc** [reply]
gotta get that beach time

**Comment by Brihalia** [comment]
**1. What is one reason you're in this class (outside of requirements)?**
My name is Brihalia Kenton. I am a mechanical engineering major. I am taking this class because I think it will be useful in understanding and designing systems to solve different problems.
**2. What is your favorite Summer activity?**
My favorite summer activity is having family cookouts because there is good food and good company. 
**3. Share a snippet of Python code and tell us what it does***
The code is a dice roll that chooses a random number from 0 to 6.
```python
import random
def dice_roll():
  n=random.randrange(0, 7,1 )
  print('Your number is',n)
dice_roll()
```

**Reply by cooperrc** [reply]
Food + family == good times :100: 

latest implementation of `random` is 

```python
from numpy.random import default_rng
rng = default_rng()

rng.randint(low = 1, high = 6, size = 1)
```

then you could have N-sided dice with as many rolls as you need e.g. `size = 3` for 3 dice.

**Comment by aiden-hasson** [comment]
Hi everyone I'm Aiden Hasson. I'm a rising senior, Mechanical Engineering major.

**What is one reason you're in this class (outside of requirements)?**
Outside of this class being a requirement, I'm taking ME3255 to further my knowledge with coding due to the fact that it is becoming such a key aspect in the digital world and provides versatility within the ME field.
**What is your favorite Summer activity?**
Right now I am doing the 75 hard challenge so I believe my favorite summer activity would be exercising; other than that I love going to the beach and tanning.
**Share a snippet of Python code and tell us what it does***
My code gives random facts.

`import randfacts as rf`

`print('Random Facts')`
`x=' '`

`while x==' ':`
`print('Press Enter for Fact')`
`x = input()`
`fact = rf.get_fact()`
`print(fact)`

**Reply by cooperrc** [reply]
I hadn't seen the [hard-75](https://health.clevelandclinic.org/75-hard-challenge-and-rules) before. Good luck!

I like the `randfacts` library. 

for those interested, you can install with:

```python
! pip install randfacts
```

**Comment by PatricioSalomon** [comment]
**What is one reason you're in this class (outside of requirements)?**
Hey everyone, my name is Patricio Salomon and I'm studying Mechanical Engineering and Industrial Design. I'm looking forward to this class because I'm on a team that works on topology optimization research and I'm excited to finally have an understanding of the computational calculations behind the optimization programs being developed

**What is your favorite Summer activity?**
Having cookouts outdoors with friends and getting a chance to visit my extended family

**Share a snippet of Python code and tell us what it does**

```python
# Call for input variables
Length = float(input("Enter the length of the rectangular prism (cm): "))
Width = float(input("Enter the width of the rectangular prism (cm): "))
Height = float(input("Enter the height of the rectangular prism (cm): "))

# Define volume and surface area functions
def surface_area(Length, Width, Height):
    return 2 * (Length * Width + Length * Height + Width * Height)

def volume(Length, Width, Height):
    return Length * Width * Height

# Display results
print("Surface area of the rectangular prism:", surface_area(Length, Width, Height),"square centimeters")
print("Volume of the rectangular prism:", volume(Length, Width, Height),"cubic centimeters")
```
This code uses user inputs for the length, width, and height of a rectangular prism to solve for it's surface area and volume

**Reply by cooperrc** [reply]
Topology optimization is super interesting, welcome!

**Comment by Vxx24001** [comment]
Hi everyone! My name is Simon Basta, and I'm a student at George Mason University in VA. I'm excited to be part of this summer session.

1. What is one reason you're in this class (outside of requirements)?

I'm here because I'm eager to improve my coding skills, especially in Python. Learning to code will greatly benefit my understanding of engineering concepts and prepare me for future projects.

2. What is your favorite Summer activity?

I enjoy going to the gym, spending time with my family, exploring new places through travel, and actively searching for internship opportunities. These activities keep me motivated and engaged during the summer months.

3. Share a snippet of Python code and tell us what it does

Here's a basic Python program that checks if a number is even or odd:

``` python
# Function to check if a number is even or odd
def check_even_odd(num):
    if num % 2 == 0:
        return f"{num} is even"
    else:
        return f"{num} is odd"

# Example usage
number = 7
result = check_even_odd(number)
print(result)
```

This Python code snippet checks whether a number is even or odd using a conditional expression (if-else) and prints the result directly. When you run this with number = 7, it will output: 7 is odd

**Reply by cooperrc** [reply]
Welcome to our UConn course! That's awesome, we have x5 people from George Mason. If you haven't met each other, you should have local meetup.

**Comment by Adam-Skarre** [comment]
> Greetings!

 
 > 1. What is one reason you're in this class (outside of requirements)?

Hello everybody, my name is Adam Skarre and I study mechanical engineering. I am here to take computational mechanics and learn how to implement python to solve engineering based problems and understand computational problems.

> 2. What is your favorite Summer activity?

A trick I used for relaxation was spending time in the sun, specifically poolside because I just opened my pool!

> 3. Share a snippet of Python code and tell us what it does*
A " cool " python code: This code simply adds two numbers and gets the sum. Simple, yet effective!
> 
> > * use `` ```python ``` `` brackets to make a copy-able code cell e.g.
> > ```
> > 

> >  # Define two numbers
```python
number1 = 3
number2 = 4

# Calculate the sum
sum_result = number1 + number2
```python
# Print the result
print("The sum is:", sum_result)
> > ```

**Comment by carol-kheng** [comment]
> 1. What is one reason you're in this class (outside of requirements)?

Hello my name is Carol Kheng, I am a rising senior majoring in mechanical engineering. I am interested in taking this course to further my interest engineering applications of coding and improve my skills in languages like python and MATLAB.  

> 2. What is your favorite Summer activity?

My favorite summer activity is going new places and trying different food. Last summer I went to Greece where I shadowed doctors and volunteered at local clinics. I love seeing different cultures and learning more about its history!

> 3. Share a snippet of Python code and tell us what it does*

```python
    def predict(self):
        self.changes_saved = False
        self.image1.save("temp.png")
        img = PIL.Image.open("temp.png")
        img.thumbnail((50, 50))
        img.save("predictshape.png", "PNG")

        img = cv.imread("predictshape.png")[:, :, 0]
        img = img.reshape(2500)
        prediction = self.clf.predict([img])

        tkinter.messagebox.showinfo("Drawing Classifier", f"Drawing is {prediction}")
```
This is a snippet of code from a simple draw pad ai detector I made. It uses different classifier models and gathers data from the user's drawings and makes predictions based on the class names it was trained with.

**Comment by rayan9732** [comment]
Hi my name is Rayan.
What is one reason you're in this class (outside of requirements)?
To better my skills in python
What is your favorite Summer activity?
Vacationing in tropical climates
Share a snippet of Python code and tell us what it does*
name = 'Rayan'
if name == 'Rayan':
        print("Hello Rayan")
else:
       print("Don't know you bro")
This code doesnt know you if your name isnt rayan

**Comment by Roe20003** [comment]
Hello My name is Roudy 
What is one reason you're in this class (outside of requirements)? - I am in this class to help with mud coding skills and further python knowledge. 

What is your favorite Summer activity? - going to car meets and going to my home country.

Share a snippet of Python code and tell us what it does* - 
a = 5
b = 5
result = a + b
print(result) 
10

This code demonstrates how to add on python as it can be used as a calculator.
